### ISSUE TITLE
Slack notification not working for env property

### ISSUE THREAD

**Issue by ramaboggarapu** [travis-ci/travis-ci#6387]
I have an SLACK_TOKEN environment variable defined for my travis project and used the SLACK_TOKEN env variable in my .travis.yml file.

```
notifications:
  slack: ${SLACK_TOKEN}
```

Notifications are not sent to slack.

I expect to have environment variable support for slack notifications.

**Comment by random6886** [comment]
Has there been any progress on this? Is currently the only way to use the encrypted token solution and not the environmental variables?

**Comment by joshk** [comment]
I'm sorry but this is not possible. 

As we are looking at moving notification settings to the UI, this feature will not be added to the roadmap.

I'm going to close this issue for the time being.

Thank you for taking the time to suggest this.

**Comment by robpark** [comment]
Moving notification settings to the UI? I love setting the Slack token / channel in the .travis.yml and that it's encrypted. 😦

**Comment by jamesrgrinter** [comment]
Whilst it may be "easy" (for some value of easy) to have the settings, and notification tokens, individually in each repo's .travis.yml, here's why I'd expect to see it as a central setting:

1. If I have an access token compromise, or my policy requires it to be regenerated periodically, I should be able to change it with one setting update.
1. I should be able to ensure, by a central policy, that all the Travis operations that I want to be notified are indeed notified. If it is my intention for every build success and failure to send a notification then It should not be possible for a developer commit to disable them.

**Comment by keremcankabadayi** [comment]
Hi there all, I'm from future:) I coded workaround for this. Here my problem and solution, maybe someone may see this.

I have flutter app. I needed to build apk and shipped that apk somehow. I used some curl uploadable hosting. But I needed that hosting url to send notification to my test team on Slack.

I set APK_URL when upload was done. Trick is here. I coded node app to send notification and used shelljs to get build env. variable. 

https://www.npmjs.com/package/slack-notify
https://www.npmjs.com/package/shelljs

Maybe someone pushes PR, happy coding.
